cmd/tiller-proxy/internal/handler/handler.go

```diff
@@ -171,7 +171,9 @@ func (h *TillerProxy) CreateRelease(w http.ResponseWriter, req *http.Request, pa
 	if !h.DisableAuth {
 		manifest, err := h.ProxyClient.ResolveManifest(params["namespace"], chartDetails.Values, ch)
 		if err != nil {
-			response.NewErrorResponse(errorCode(err), err.Error()).Write(w)
+			// To resolve the manifest we are contacting tiller that may return an error.
```

Comment:
Just my 2 cents: should we only use 500 to indicate that it was a server error in this service? In this case, this service processed the request fine, but Tiller returned the error, I think it can be considered as unprocessable and not a server error.

Reply:
One case here is when a user is installing a chart that is broken by default (i.e. tiller will complain on a bare `helm install`), and it might be an upstream chart that the user is trying to deploy. In this case, it's not really the fault of the user or their inputs, so is it really a bad request? That said, Unprocessable Entity !== Bad Request (400). That said, we're all probably reading too much into this and Unprocessable Entity will work fine here.

Reply:
I still think that by default we should return `unknown` (or 500) error except in the `Tiller minimum version requirement` whitelisted case. That way we are in the safe side by never masking possible tiller errors (panics, crashes, non-ready) as `Unprocessable Entity` errors.

Reply:
The other option would be to actually change the UI to handle this specific case, and show that there was an error upgrading the app. At the end of the day that's something we should do even if Tiller or Tiller proxy are down (503).

I'd personally prefer to not to change anything here in the backend and explore this other option but up to you.

Reply:
I don't like the idea of whitelisting the specific message for the tiller version. As you say we could just improve how we represent errors to the users in the frontend. I will revert this change and open a new issue to tackle that.

Reply:
#856